### PR TITLE
[PPP-5163] Vulnerable Component: snakeyaml - part 2 ( Non OSGi upgrade)

### PR DESCRIPTION
**Pull request by NJtwentyone**
- replace 'org.hitachivantatar:safeyaml' with 'org.yaml:snakeyaml:[2.0,)'

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-war,extensions,repository
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 44m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-war,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2028**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5723/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/pom.xml [hunk 1]
```diff
@@ -208,9 +208,9 @@
       <scope>compile</scope>
     </dependency>
     <dependency>
-      <groupId>org.hitachivantara</groupId>
-      <artifactId>safeyaml</artifactId>
-      <version>${safeyaml.version}</version>
+      <groupId>org.yaml</groupId>
```
Comment (NJtwentyone) [inline repository/pom.xml:211]: `Yaml()` is called here: https://github.com/pentaho/pentaho-platform/blob/156fb63fddacda4de89be2f2c7882b7e50f39f62/repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/MagicAceDefinition.java#L141

File: assemblies/pentaho-war/pom.xml [hunk 2]
```diff
@@ -840,8 +840,9 @@
       </exclusions>
     </dependency>
     <dependency>
-      <groupId>org.hitachivantara</groupId>
```
Comment (NJtwentyone) [inline assemblies/pentaho-war/pom.xml:843]: The library snakeyaml-2.2.jar ( "org.yaml:snakeyaml:2.2") replaces the previous interations of safeyaml-1.34.1.jar ( "org.hitachivantara:safeyaml" and "com.konloch:safeyaml") and snakeyaml-1.33.jar (""org.yaml:snakeyaml:1.33")

The artifact `snakeyaml.jar` is a runtime dependency on `pentaho-platform-exentions.jar` and `pentaho-platform-repository.jar`

For pentaho reporting the "library" folders:
- PS-CE (before)
  - server/pentaho-server/tomcat/webapps/pentaho/WEB-INF/lib/pentaho-platform-extensions.jar
  - **server/pentaho-server/tomcat/webapps/pentaho/WEB-INF/lib/safeyaml.jar**
- PS-CE (after)
  - server/pentaho-server/tomcat/webapps/pentaho/WEB-INF/lib/pentaho-platform-extensions.jar
  - **server/pentaho-server/tomcat/webapps/pentaho/WEB-INF/lib/snakeyaml.jar**

PS-CE feeds into PS-EE so  no need to duplicate this configuration in PS-EE pom.xmls

**Review by singletonc — APPROVED** [review]

**Review by rmansoor — APPROVED** [review]

**Pull request merged**